pkg/cmd/admin/router/router.go

```diff
@@ -325,6 +325,11 @@ func generateSecretsConfig(cfg *RouterConfig, kClient *kclient.Client,
 		mounts = append(mounts, mount)
 	}
 
+	// When the user provides a DEFAULT_CERT environment variable it must be in pem format.
+	// If the default cert is not provided by the user, we generate one using a service
+	// annotation (see RunCmdRouter()), but that will have a cert and a key that we need to
+	// combine into the pem that haproxy expects
+	var certName = fmt.Sprintf("%s-certs", cfg.Name)
 	if len(defaultCert) > 0 {
 		keys, err := cmdutil.PrivateKeysFromPEM(defaultCert)
```

Comment:
describe for future readers: `if the user passes in a default certificate, extract the private key, and then stick the private key and the whole PEM in a secret to be mounted below`